### ISSUE TITLE
ReadFrom.Configuration(...) throws an exception on AWS Lambda when dynamically loading sinks

### ISSUE THREAD

**Issue by mawt** [serilog/serilog#957]
When doing something as mundane as this:

```
         var logger = new LoggerConfiguration()
                .ReadFrom.Configuration(configuration)
                .CreateLogger();
```

In a netcoreapp1.0 running as a Lambda on AWS, Serilog will trigger a null exception in `DependencyContextLoader.Load(Assembly assembly)` thanks to the code behind `DependencyContext.Default` calling `Assembly.GetEntryAssembly()` which returns null in certain conditions (see [Remarks](https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getentryassembly(v=vs.110).aspx) here). 

Please see [this issue](https://github.com/Microsoft/vstest/issues/649) more information about the issue and an [upstream patch](https://github.com/dotnet/core-setup/issues/1017).

As a result of these issues, Serilog JSON configuration cannot currently be used for logging on AWS Lambda. Is there a workaround available?

Kind regards, Matt

**Comment by nblumhardt** [comment]
Hi Matt, thanks for the report!

Just to keep the issue description informative, it's a bit more subtle than:

> Serilog cannot currently be used for logging on AWS

I think to be precise it's:

> Serilog **JSON configuration** cannot currently be used for logging on AWS **Lambda**

You can still add sinks using the typical `WriteTo` C# syntax (I know of a few Lambda projects doing this).

Do you think we should wait for the upstream patch, or should we work around it in [the JSON config provider](https://github.com/serilog/serilog-settings-configuration/blob/dev/src/Serilog.Settings.Configuration/ConfigurationLoggerConfigurationExtensions.cs#L63)?

Cheers!

**Comment by mawt** [comment]
Thanks for the quick response, I've edited the issue above for clarity.

From what I can see there isn't a way of intercepting the Assembly loading in .net core and certainly not for LTS releases like netcoreapp1.0. My guess is this problem isn't likely to go away soon.

My suggestion would be to implement a callback where users can provide a `Func<>` to implement their own Assembly resolving/loading/caching. This would be extremely useful for assemblies that live outside of the search path.

**Comment by skomis-mm** [comment]
@mawt , would using `Using` section is sufficient? 

I vote for workaround in `Serilog.Settings.Configuration` package than for waiting for the patch (because it also means reference for the newer version of the `Microsoft.Extensions.DependencyModel`). At least for the `2.x` `Serilog.Settings.Configuration`'s  branch.

**Comment by mawt** [comment]
Thank you!

**Comment by nblumhardt** [comment]
@mawt does the `-00049` package work for you? https://www.nuget.org/packages/Serilog.Settings.Configuration/2.3.1-dev-00049 - If so, let me know, we'll ship the patch release today. Thanks!

**Comment by mawt** [comment]
It works - watching logs appear in Cloudwatch as I type :)

**Comment by nblumhardt** [comment]
Great - published 2.3.1. Thanks @skomis-mm!
